lighthouse-core/report/v2/renderer/category-renderer.js

```diff
@@ -336,6 +350,86 @@ class CategoryRenderer {
   }
 
   /**
+   * Render a screenshot filmstrip and metrics bars
+   * @param {!Array<!ReportRenderer.AuditJSON>} metricAudits
+   * @param {*} artifacts
+   * @return {?Element}
+   */
+  _renderPerfTimeline(metricAudits, artifacts) {
+    if (!artifacts)
+      return;
+    const tracePass = artifacts['traces'] ? artifacts['traces']['defaultPass'] : null;
+    if (!tracePass)
+      return;
+
+    const fmp = metricAudits.find(a => a.id === 'first-meaningful-paint').result;
+    if (!fmp || !fmp['extendedInfo'])
+      return;
+
+    const tti = metricAudits.find(a => a.id === 'time-to-interactive').result;
+    if (!tti || !tti['extendedInfo'])
+      return;
+
+    const navStart = fmp['extendedInfo']['value']['timestamps']['navStart'];
+    const markers = [
```

Comment:
Basically I'm proposing to move the logic that collects the metrics from pwmetrics-events.js into an audit instead.

audits/metrics.js
```js
const TTFI = require('./first-interactive')

class Metrics extends Audit {
  static audit(artifacts) {
    const traceOfTab = await artifacts.requestTraceOfTab()
    const fmp = await FMP.audit(artifacts)
    const firstInteractive = await TTFI.audit(artifacts)

    return {
      extendedInfo: {
        value: [
           {name: 'first-interactive', timeInMs: firstInteractive.extendedInfo.value.timings.firstInteractive, timestamp: ...},
           {name: 'on-load', timeInMs: traceOfTab.timings.onLoad, timestamp: ...},
           {name: 'trace-end', timeInMs: traceOfTab.timings.traceEnd, timestamp: ...},
           ...
        ]
      }
    }
  }
}
```

pwmetrics-events.js
```js
class Metrics {
  ...
  metricsDefinitions(auditResults) {
    return auditResults['metrics'].extendedInfo.value;
  }
}
```